### PR TITLE
v3 fast follows

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA:
- https://jira.mongodb.org/browse/DOCSP-44565
- https://jira.mongodb.org/browse/DOCSP-44569
- https://jira.mongodb.org/browse/DOCSP-44544

### Staging Links
https://deploy-preview-265--mongodb-docs-csharp.netlify.app/upgrade/v3
https://deploy-preview-265--mongodb-docs-csharp.netlify.app/compatibility

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Review by mayaraman19 — APPROVED** [review]
LGTM

File: source/upgrade/v3.txt [hunk 1]
```diff
@@ -61,6 +61,17 @@ Version 3.0 Breaking Changes
...
+- The driver drops support for .NET Core 2.x and .NET Framework 4.6. You must upgrade to
+  .NET Core 3.x or later, or .NET Framework 4.7.2 or later.
+
+  To learn more about the compatibility between .NET/C# driver versions
+  and .NET versions, visit the
+  :ref:`Compatibility <csharp-compatibility-tables>` page.
+
+- The v1.x legacy API, which was deprecated in v2, has been removed. If the driver
+  provides a replacement for a deprecated API, the compiler
+  messages in v2.x will display it.
```
Comment (JamesKovacs) [inline source/upgrade/v3.txt:73]: There are two separate issues here.

First the legacy v1.x API was removed. That was contained in the legacy shim `mongocsharpdriver`. If you're using those v1.x APIs, you need to migrate to the new API and we don't provide any upgrade guidelines. Very few users are still on the old v1.x API and thus we don't provide upgrade guidance. (We don't need to say that. Just FYI.)

Second the v2.x API contained a number of deprecations, which have been removed in 3.0.0. The 2.30.0 driver contains guidance in the form of deprecation warnings including the new APIs to use.

**Review by JamesKovacs — CHANGES_REQUESTED** [review]
Clarification requested around removal of the legacy shim `mongocsharpdriver` and the v2.x API deprecations.

File: source/upgrade/v3.txt (hunk 1 above)
Comment (mongoKart) [reply source/upgrade/v3.txt:73]: I appreciate the clarification. It was hard to find specifics about this.

**Review by JamesKovacs — APPROVED** [review]
LGTM

**Pull request merged**